packages/navigator/src/browser/navigator-model.ts

```diff
@@ -35,4 +36,60 @@ export class FileNavigatorModel extends FileTreeModel {
         }
     }
 
+    /**
+     * Reveals node in the navigator by given file uri.
+     *
+     * @param targetFileUri uri to file whch should be revealed in the navigator
+     * @returns true if file with given uri was revealed, false otherwise
+     */
+    async revealFile(targetFileUri: URI): Promise<ITreeNode | undefined> {
+        const navigatorNodeId = targetFileUri.toString();
+        let node = this.getNode(navigatorNodeId);
+
+        // success stop condition
+        // we have to reach workspace root because expanded node could be inside collapsed one
+        if (this.root === node) {
+            if (IExpandableTreeNode.is(node)) {
+                if (!node.expanded) {
+                    await this.expandNodePromise(node);
+                }
+                return node;
+            }
+            // shouldn't happen, root node is always directory, i.e. expandable
+            return undefined;
+        }
+
+        // fail stop condition
+        if (targetFileUri.path.isRoot) {
+            // file system root is reached but workspace root wasn't found, it means that
+            // given uri is not in workspace root folder or points to not existing file.
+            return undefined;
+        }
+
+        if (await this.revealFile(targetFileUri.parent)) {
+            if (node === undefined) {
+                // get node if it wasn't mounted into navigator tree before expansion
+                node = this.getNode(navigatorNodeId);
+            }
+            if (IExpandableTreeNode.is(node) && !node.expanded) {
+                await this.expandNodePromise(node);
+            }
+            return node;
+        }
+        return undefined;
+    }
+
+    protected expandNodePromise(node: IExpandableTreeNode): Promise<void> {
+        return new Promise(resolve => {
+            // onExpansionChanged event doesn't work here because it is fired before actual expanding
+            const subscribtion = this.onNodeRefreshed(expandedNode => {
+                if (node.expanded && expandedNode.id === node.id) {
+                    subscribtion.dispose();
+                    resolve();
+                }
+            });
+            this.expandNode(node);
```

Comment:
please promisifiy `TreeModel.expandNode` and `TreeModel.refresh` instead of using `onNodeRefreshed`

Reply:
Yes, I just decided to push fixes